pkg/stanza/operator/input/udp/udp.go

```diff
@@ -101,6 +113,14 @@ func (c Config) Build(logger *zap.SugaredLogger) (operator.Operator, error) {
 		resolver = helper.NewIPResolver()
 	}
 
+	if c.AsyncConfig == (UdpAsyncConfig{}) {
```

Comment:
Typically for optional fields we make it a pointer and then check `== nil`
```suggestion
	if c.AsyncConfig == nil {
```